### DISCUSSION TITLE
New version added libzstd.arm.32.so etc. to the project

### DISCUSSION THREAD

**Discussion by ashokgudur** [ravendb/ravendb#13357] [General]
Hi,
My project is .NET Standard Class Library.
When I upgraded to newer version v5.1 there are extra files added as show in the attached screenshot.
Why? I didn't see them before. That makes my project looks bit awkward. :)

<img width="226" alt="Screenshot 2021-12-24 225358" src="https://user-images.githubusercontent.com/3616094/147367182-d5d76415-3ecb-4675-a1b5-b95f7b77d719.png">

**Comment by ayende** [comment]
Just to note, RavenDB 5.1 isn't a new version, the new version is 5.3. The 5.1 is not supported at this time.

The actual behavior is consistent with 5.3 as well. This will happen when you are using either the `Raven.TestDriver` or `Raven.Embedded`

The underlying reason is that we support all those architectures, and we make use of native libraries for some features, as such, we need to provide them for the end application to use as well.

**Reply by ashokgudur** [reply]
Well I mean v5.3.1 only. It was a typo. 
Anyway thanks for the reply. 

So, those file references are required and should be copied to output directory when deployed – Right?

No issues. 
Thanks.

**Reply by ayende** [reply]
Those are required when you are deploying _for that architecture_.
If you are deploying to 32 bits arm, yes. If you aren't, you can drop that.

**Reply by PureKrome** [reply]
👋🏻 Hi @ayende 

> either the `Raven.TestDriver` or `Raven.Embedded` 

I'm just using `Raven.Client` v 5.3.1 (latest, at the time of me posting this) and I get the following:

![image](https://user-images.githubusercontent.com/899878/147803092-23acea43-08a6-4f74-a685-0abf21f2a31b.png)

So this is a `NET6.0` project and it's listing all the files.

Is this correct?

**Reply by ayende** [reply]
Sorry, that is correct, this is TCP compression feature, which we now use on Subscriptions as well. 
Sorry, that would be on the standard client too

**Reply by PureKrome** [reply]
Ah, okay thanks.

Why does this have to be here instead of in some nuget package? I can see that these are all architecture specific but ... maybe my OCD kicks in and i feel like this makes me cringe. Yes, it's purely an aesthetic thing. Like, NuGet offers 'architecture specific folders' (ref: https://docs.microsoft.com/en-us/nuget/create-packages/supporting-multiple-target-frameworks#architecture-specific-folders) .. could this be leveraged?

> These assemblies will only be available at runtime, so if you want to provide the corresponding compile time assembly as well then have AnyCPU assembly in /ref/{tfm} folder.

**Reply by ppekrol** [reply]
Fixed in: https://github.com/ravendb/ravendb/pull/13388
